src/common/framework/test_group.ts

```diff
@@ -194,29 +195,25 @@ class TestBuilder<F extends Fixture, P extends {}, SubP extends {}> {
     }
   }
 
-  params<NewP extends {}>(casesIterable: Iterable<NewP>): TestBuilder<F, NewP, SubP> {
+  params(casesIterable: Iterable<{}>): TestBuilder {
     return this.cases(casesIterable);
   }
 
-  cases<NewP extends {}>(casesIterable: Iterable<NewP>): TestBuilder<F, NewP, SubP> {
+  cases(casesIterable: Iterable<{}>): TestBuilder {
     assert(this.caseParams === undefined, 'test case is already parameterized');
-    const newSelf = (this as unknown) as TestBuilder<F, NewP, SubP>;
-    newSelf.caseParams = Array.from(casesIterable);
-
-    return newSelf;
+    this.caseParams = Array.from(casesIterable);
+    return this;
   }
 
-  subcases<NewSubP extends {}>(specs: (_: P) => Iterable<NewSubP>): TestBuilder<F, P, NewSubP> {
+  subcases(specs: (_: {}) => Iterable<{}>): TestBuilder {
```

Comment:
does this mean we lose the type information on `params` when doing

.subcases(params => {
})

?
I found it kind of nice

Reply:
It still works, because the test code sees it as a `TestBuilderWithCases<F, P>`, not as a `TestBuilder`.